common/src/com/thoughtworks/go/remote/work/BuildAssignment.java

```diff
@@ -1,47 +1,65 @@
-/*************************GO-LICENSE-START*********************************
- * Copyright 2014 ThoughtWorks, Inc.
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- *************************GO-LICENSE-END***********************************/
+ */
 
 package com.thoughtworks.go.remote.work;
 
 import java.io.File;
 import java.io.Serializable;
+import java.util.ArrayList;
 import java.util.List;
 
+import com.thoughtworks.go.domain.MaterialRevision;
 import com.thoughtworks.go.domain.builder.Builder;
 import com.thoughtworks.go.domain.JobPlan;
 import com.thoughtworks.go.domain.MaterialRevisions;
 import com.thoughtworks.go.domain.buildcause.BuildCause;
+import com.thoughtworks.go.domain.materials.Modification;
 import com.thoughtworks.go.util.command.EnvironmentVariableContext;
 
 public class BuildAssignment implements Serializable {
     private final File buildWorkingDirectory;
     private final List<Builder> builders;
     private final JobPlan plan;
-    private final BuildCause buildCause;
     private final EnvironmentVariableContext initialContext = new EnvironmentVariableContext();
+    private final MaterialRevisions materialRevisions = new MaterialRevisions();
+    private final String approver;
 
     private BuildAssignment(BuildCause buildCause, File buildWorkingDirectory, List<Builder> builder, JobPlan plan) {
-        this.buildCause = buildCause;
         this.buildWorkingDirectory = buildWorkingDirectory;
         this.builders = builder;
         this.plan = plan;
+        for (MaterialRevision materialRevision : buildCause.getMaterialRevisions()) {
+            ArrayList<Modification> modifications = new ArrayList<>();
+            for (Modification modification : materialRevision.getModifications()) {
+                Modification modificationForAgent = new Modification(modification);
+                modificationForAgent.unsetModifiedFiles();
+                modifications.add(modificationForAgent);
```

Comment:
Move this to a copy constructor with a flag that constructs Modification object without copying the files so that you can get rid of the ```unsetModifiedFiles``` method.